python/pyspark/testing/utils.py

```diff
@@ -209,3 +220,45 @@ def check_error(
         self.assertEqual(
             expected, actual, f"Expected message parameters was '{expected}', got '{actual}'"
         )
+
+
+def assertSparkSchemaEquality(
+    s1: Optional[Union[AtomicType, StructType, str, List[str], Tuple[str, ...]]],
+    s2: Optional[Union[AtomicType, StructType, str, List[str], Tuple[str, ...]]],
+):
+    if s1 != s2:
+        msg = "Schemas are different"
+        raise AssertionError(msg)
+
+
+def assertSparkDFEquality(
```

Comment:
We don't necessarily have to follow how we do in Scala side but Scala has a similar internal function called `checkAnswer` at `org.apache.spark.sql.QueryTest`. You might want to refer how it's done there.